x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/watch/WatchStatusIntegrationTests.java

```diff
@@ -54,11 +55,21 @@ public void testThatStatusGetsUpdated() {
 
         String lastChecked = source.getValue("status.last_checked");
         assertThat(lastChecked, WatcherTestUtils.isSameDate(getWatchResponse.getStatus().lastChecked()));
-
+        assertThat(getWatchResponse.getStatus().lastChecked(), isMillisResolution());
         // not started yet, so both nulls
         String lastMetCondition = source.getValue("status.last_met_condition");
         assertThat(lastMetCondition, is(nullValue()));
         assertThat(getWatchResponse.getStatus().lastMetCondition(), is(nullValue()));
     }
 
+    private Matcher<ZonedDateTime> isMillisResolution() {
+        return new FeatureMatcher<ZonedDateTime,Boolean>(equalTo(true), "has millisecond precision", "precission") {
+            @Override
+            protected Boolean featureValueOf(ZonedDateTime actual) {
+                //if date has millisecond precision its nanosecond field will be rounded to millis (equal millis * 10^6)
+                return actual.getNano() == actual.get(ChronoField.MILLI_OF_SECOND) * 1000_000;
```

Comment:
alternatively just `actual.getNano() % 1000 == 0` - not sure if it is more readable to be honest, so feel free to ignore me  :-)